libs/blocks/global-navigation/global-navigation.js

```diff
@@ -505,8 +505,46 @@ class Gnav {
     };
 
     const onAnalyticsEvent = (data) => {
-      // TODO: update based on requirements
-      console.log('analytics data', data);
+      if (!data) return;
+
+      const getInteraction = () => {
+        const { event: { type, subtype } = {}, source: { name } = {} } = data;
+        const contentName = data.content?.name;
+        switch (`${name}|${type}|${subtype}|${contentName || ''}`) {
+          case 'profile|click|sign-in|':
+            return `Sign in|gnav|${getExperienceName()}|unav`;
+          case 'profile|render|component|':
+            return `Account|gnav|${getExperienceName()}`;
+          case 'profile|click|account|':
+            return `View Account|gnav|${getExperienceName()}`;
+          case 'profile|click|sign-out|':
+            return `Sign out|gnav|${getExperienceName()}|unav`;
```

Comment:
Maybe it's a micro-optimisation, but: could we cache `getExperienceName()` in a variable outside this method so it's not called for each analytics event? It shouldn't be an expensive operation, but it's easy enough to avoid multiple calls to it.

Reply:
Nice optimisation, made the change 👍 It seems so obvious after you pointed it out